src/elysium/wallet.cpp

```diff
@@ -237,9 +313,12 @@ void Wallet::OnMintAdded(
     } else {
 
         // 2. try to recover new mint
-        if (mintWallet.TryRecoverMint(
-            id, SigmaMintChainState(block, group, idx)
-        )) {
+        SigmaMintChainState chainState(block, group, idx);
+        if (mintWalletV0.TryRecoverMint(id, chainState)) {
+            LogPrintf("%s : Found new legacy mint when try to recover\n", __func__);
+        }
+
+        if (mintWalletV1.TryRecoverMint(id, chainState)) {
```

Comment:
The mint should not belongs to more than one wallet. So better to chain this condition with the above condition.

Reply:
Make sense